lib/ansible/module_utils/ansible_tower.py

```diff
@@ -42,33 +42,37 @@ def tower_auth_config(module):
     '''tower_auth_config attempts to load the tower-cli.cfg file
     specified from the `tower_config_file` parameter. If found,
     if returns the contents of the file as a dictionary, else
-    it will attempt to fetch values from the module pararms and
-    only pass those values that have been set.
+    it will attempt to pop the values from the module parameters.
+    It will only pass the values that might have been set and will clean
+    the dictionary from the tower keys created in tower_argument_spec,
+    so that when this function completes, the module.params dictionary will
+    not contain any keys related to the tower authentication.
     '''
+    tower_auth_args = [
+        'tower_host',
+        'tower_username',
+        'tower_password',
+        'tower_verify_ssl',
+        'tower_config_file',
+    ]
     config_file = module.params.pop('tower_config_file', None)
     if config_file:
         config_file = os.path.expanduser(config_file)
         if not os.path.exists(config_file):
             module.fail_json(msg='file not found: %s' % config_file)
         if os.path.isdir(config_file):
             module.fail_json(msg='directory can not be used as config file: %s' % config_file)
-
+        for arg in tower_auth_args:
+            module.params.pop(arg, None)
         with open(config_file, 'rb') as f:
             return parser.string_to_dict(f.read())
     else:
         auth_config = {}
-        host = module.params.pop('tower_host', None)
-        if host:
-            auth_config['host'] = host
-        username = module.params.pop('tower_username', None)
-        if username:
-            auth_config['username'] = username
-        password = module.params.pop('tower_password', None)
-        if password:
-            auth_config['password'] = password
-        verify_ssl = module.params.pop('tower_verify_ssl', None)
-        if verify_ssl is not None:
-            auth_config['verify_ssl'] = verify_ssl
+        for arg in tower_auth_args:
+            argk = arg.replace('tower_', '', 1)
+            argv = module.params.pop(arg, None)
```

Comment:
`value` could be used instead of `argv`. `argk` is only required when `argv is not None` and could be moved in the `if` block.